test/mpi/autogen.sh

```diff
@@ -31,6 +31,10 @@ echo "Running autoreconf in dtpools"
 # Create and/or update the f90 tests
 printf "Create or update the Fortran 90 tests derived from the Fortran 77 tests... "
 for dir in f77/* ; do
+    if [ $dir = 'f77/profile' ] ; then
+        # skip profiling test for "use mpi"
+        continue
+    fi
```

Comment:
Not sure I understand why the compiler won't allow the profiling routines. Could you explain more?

Reply:
In `test/mpi/f77/profile/profile1f.f`, it includes a `subroutine mpi_send`. Now that `mpi_send` is listed in the `mpi.mod` interface, the compiler doesn't allow a duplicated one.

Reply:
I see. The MPI Forum suggests renaming the interface when using the module inside the subroutine definition in order to workaround this problem. See MPI-4.0 pg 803. It does work in my limited testing, though we'll either have to modify the test conversion script or just commit this test by itself.
```fortran
       subroutine mpi_send( smsg, count, dtype, dest, tag, comm, ierr )
       use mpi, my_noname => mpi_send
```

Reply:
The fix works for gfortran-4.9+. But gfortran-4.8 doesn't support ignore-tkr, thus mpi.mod doesn't contain routines as `mpi_send`, the above fix won't compile. I added a `sed` in configure to patch it if that is the case.

Reply:
Note: the `sed` in configure potentially modifies a distributed file, thus it is a bit hairy. Fortran doesn't do macros, and I don't have a good solution to do autoconf `.in` substitution. So that leaves the solution of moving `f77tof90` to a config command in `configure`. 

I'll go ahead merge this PR. If we decide to move `f77tof90` to configure, we can patch it later.

Reply:
Oh, the complication is, `f77tof90` will generate automaker files, so it will mess up the autoconf flow. So we'll need find another solution